pkg/httpclient/client.go

```diff
@@ -145,5 +156,33 @@ func (c *Client) NewRequest(method, path string, body io.Reader, opts ...Request
 // policy (such as redirects, cookies, auth) as configured on the
 // client.
 func (c *Client) Do(req *http.Request) (*http.Response, error) {
-	return c.baseClient.Do(req)
+	var dumpBody bool
+	var logFn func(format string, args ...interface{})
```

Comment:
What about `logFn := c.logger. Infof ` and then the `if c.logger != nil && c.logger.Level >= logrus.DebugLevel {`?